sqlmock.go

```diff
@@ -102,6 +114,13 @@ func (c *sqlmock) MatchExpectationsInOrder(b bool) {
 	c.ordered = b
 }
 
+func (c *sqlmock) AllowRepeatedExpectationMatching(b bool) {
+	if b {
+		c.ordered = false
```

Comment:
though, lets leave it as a requirement to set it explicitly. implicit option changes are magic and hard to understand. and that will be it. good job, looks like a great feature